recipes/zlib/all/conanfile.py

```diff
@@ -29,7 +29,7 @@ class ZlibConan(ConanFile):
 
     @property
     def _is_mingw(self):
-        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+        return self.settings.os == "Windows" and self.settings.compiler in ["gcc", "clang"]
```

Comment:
It would break clang-cl.

I've proposed a helper in conan client https://github.com/conan-io/conan/pull/12678, but it has been rejected.

Reply:
It got rejected because it was a conan 1.x issue. If this is common throughout conan, I think there must be a generic and centralized solution to it. Especially since these days, if someone is using a gcc-derivative on Windows, the most probable choice would be `clang` because of `libc++`.